pkg/network/go/lutgen/run.go

```diff
@@ -0,0 +1,301 @@
+	// be skipped when it is known that
+	// they don't cause different lookup table values.
+	allVersions := []goversion.GoVersion{}
+	for _, rawVersion := range allRawVersions {
+		if version, ok := goversion.Parse(fmt.Sprintf("go%s", rawVersion)); ok {
+			if version.Beta != 0 ||
+				version.RC != 0 ||
+				version.Proposal != "" ||
+				!version.AfterOrEqual(g.MinGoVersion) {
+				continue
+			}
+
+			// Check if the version's major, minor pair falls under the set
+			// of "quick" Go versions where non-zero revisions are skipped.
+			if g.MaxQuickGoVersion != nil &&
+				g.MaxQuickGoVersion.AfterOrEqual(version) &&
+				version.Rev > 0 {
+				continue
+			}
+
+			allVersions = append(allVersions, version)
+		}
+	}
+
+	return allVersions, nil
+}
+
+func (g *LookupTableGenerator) addBinary(binary Binary) {
+	g.allBinariesMu.Lock()
+	defer g.allBinariesMu.Unlock()
+
+	g.allBinaries = append(g.allBinaries, binary)
+}
+
+func (g *LookupTableGenerator) getAllBinaries() []Binary {
+	g.allBinariesMu.Lock()
+	defer g.allBinariesMu.Unlock()
+
+	newSlice := make([]Binary, len(g.allBinaries))
+	copy(newSlice, g.allBinaries)
+	return newSlice
+}
+
+func (g *LookupTableGenerator) getCommand(ctx context.Context, version goversion.GoVersion, arch string) *exec.Cmd {
+	versionStr := versionToString(version)
+	outPath := filepath.Join(g.OutDirectory, fmt.Sprintf("%s.go%s", arch, versionStr))
+
+	modFile, err := g.createFakeGoMod(version, arch)
+	if err != nil {
+		log.Printf("error creating go.mod file for (Go version, arch pair) (go%s, %s): %v", versionStr, arch, err)
+		return nil
+	}
+
+	// Store the binary struct in a list so that it can later be opened.
+	// If the command ends up failing, this will be ignored
+	// and the entire lookup table generation will exit early.
+	g.addBinary(Binary{
+		Path:            outPath,
+		GoVersion:       version,
+		GoVersionString: versionStr,
+		Architecture:    arch,
+	})
+
+	command := exec.CommandContext(
+		ctx,
+		"go",
+		"build",
+		"-o", outPath,
+		"-modfile", modFile,
+		"--",
+		g.TestProgramPath,
+	)
+
+	// Set the GOPATH and GOCACHE variables.
+	// Make sure to resolve the absolute path of install directory first.
+	installDirectoryAbs, err := filepath.Abs(g.InstallDirectory)
+	if err != nil {
+		log.Printf("error install directory at %q: %v", g.InstallDirectory, err)
+		return nil
+	}
+	command.Env = append(command.Env, fmt.Sprintf("%s=%s", "GOPATH", filepath.Join(installDirectoryAbs, "build-gopath")))
+	command.Env = append(command.Env, fmt.Sprintf("%s=%s", "GOCACHE", filepath.Join(installDirectoryAbs, "build-gocache")))
+
+	return command
+}
+
+// createFakeGoMod creates the go.mod file that the `go build` command should use
+// when compiling the test program.
+// This is needed to prevent the test program compilation from using the
+// `go.mod` file in the root of this repository.
+func (g *LookupTableGenerator) createFakeGoMod(version goversion.GoVersion, arch string) (string, error) {
```

Comment:
I think in theory you can just use `GO111MODULE=off` when you call `go build` for the test binary

Reply:
Good point; I removed the fake go.mod file and passed in that to the build commands.